src/oemof/solph/flows/_non_convex_flow_block.py

```diff
@@ -469,24 +471,25 @@ def min_downtime_rule(_, i, o, t):
             """
             Rule definition for min-downtime constraints of non-convex flows.
             """
-            if self._time_step_allows_flexibility(
-                t, m.flows[i, o].nonconvex.max_up_down, m.TIMESTEPS.at(-1)
-            ):
```

Comment:
Removing this calls for a different handling of the first time steps. Maybe, we continue to not create the constraint for the first N time steps so that the first time of switching down can continue to be our marker.

Reply:
or the first timesteps have to be fixed and afterwards implement only the constraints of the timesteps > uptime/downtime

Reply:
Just to have this more graphic. The old implementation interpreted the minimum uptime or downtime to force the flow up or down in the beginning and in the end.

Old implementation:
![image](https://github.com/oemof/oemof-solph/assets/4871848/af7d6130-e358-42e5-9fd8-d9dd85b1ab70)

Current implementation:
![image](https://github.com/oemof/oemof-solph/assets/4871848/f9b1867d-48b6-4b75-ba55-ec2e791a4a6a)

Thinking about this a bit longer, there is a bug.

```python
pp2 = solph.components.Source(
    label="plant_min_up_constraints",
    outputs={
        bel: solph.Flow(
            nominal_value=10,
            min=0.5,
            max=1.0,
            variable_costs=10,
            nonconvex=solph.NonConvex(minimum_uptime=2, minimum_downtime=4, initial_status=1),
        )
    },
)
```

Results in the flow being forced active for four time steps. I think backwards compatibility wit this can be dropped, as its an obvious mistake.

![image](https://github.com/oemof/oemof-solph/assets/4871848/591c0159-13bf-4d7e-8f5e-79c006a40903)

Reply:
The goal is only to force the up or down (depending on initial_status) only in the beginning. Or is there a reason, why it has to be forced to be up / down in the end?

In  my opinion the plot of "current implementation" is showing the correct behavor.

Reply:
The "current implementation" in this branch does only force the status up/down in the beginning. However, it does so for longer than needed (third plot and https://github.com/oemof/oemof-solph/pull/1021#discussion_r1407297634).

I think, it would make sense to allow forcing the status up/down in the end of the optimisation horizon as an equivalent to the "balanced" storage. If the storage has minimum downsteps of 4, it can have 2 in the end plus 2 in the beginning. But this should just be an option. The old implementation (in v0.5.1 and before) really made no sense.